File: sdk/java/src/main/java/io/readylayer/services/HealthService.java
package io.readylayer.services;

import io.readylayer.ReadyLayerClient;
import io.readylayer.models.*;
import lombok.RequiredArgsConstructor;

import java.net.http.HttpRequest;
import java.util.concurrent.CompletableFuture;

/**
 * Service for health check operations.
 */
@RequiredArgsConstructor
public class HealthService {
    
    private final ReadyLayerClient client;
    
    /**
     * Get health status (no authentication required).
     */
    public HealthResponse getHealth() {
        HttpRequest request = client.requestBuilder("/health")
                .header("Authorization", "")
                .GET()
                .build();
        
        return client.execute(request, HealthResponse.class);
    }
    
    /**
     * Get health status asynchronously.
     */
    public CompletableFuture<HealthResponse> getHealthAsync() {
        HttpRequest request = client.requestBuilder("/health")
                .header("Authorization", "")
                .GET()
                .build();
        
        return client.executeAsync(request, HealthResponse.class);
    }
    
    /**
     * Get readiness status (no authentication required).
     */
    public ReadyResponse getReady() {
        HttpRequest request = client.requestBuilder("/ready")
                .header("Authorization", "")
                .GET()
                .build();
        
        return client.execute(request, ReadyResponse.class);
    }
    
    /**
     * Get readiness status asynchronously.
     */
    public CompletableFuture<ReadyResponse> getReadyAsync() {
        HttpRequest request = client.requestBuilder("/ready")
                .header("Authorization", "")
                .GET()
                .build();
        
        return client.executeAsync(request, ReadyResponse.class);
    }
}

File: sdk/csharp/src/Services/HealthService.cs
using ReadyLayer.SDK.Models;

namespace ReadyLayer.SDK.Services;

/// <summary>
/// Service for health check operations.
/// </summary>
public class HealthService
{
    private readonly ReadyLayerClient _client;

    internal HealthService(ReadyLayerClient client)
    {
        _client = client;
    }

    /// <summary>
    /// Checks the service health status.
    /// </summary>
    /// <param name="cancellationToken">Cancellation token.</param>
    /// <returns>The health status response.</returns>
    public async Task<HealthResponse> CheckAsync(
        CancellationToken cancellationToken = default)
    {
        return await _client.GetAsync<HealthResponse>(
            "health",
            cancellationToken: cancellationToken);
    }

    /// <summary>
    /// Checks the service readiness status including dependencies.
    /// </summary>
    /// <param name="cancellationToken">Cancellation token.</param>
    /// <returns>The readiness status response.</returns>
    public async Task<ReadyResponse> CheckReadyAsync(
        CancellationToken cancellationToken = default)
    {
        return await _client.GetAsync<ReadyResponse>(
            "ready",
            cancellationToken: cancellationToken);
    }
}
